### ISSUE TITLE
Is backwards compatibility of the database expected or not until 1.0?

### ISSUE THREAD

**Issue by dragon788** [nthdimtech/signet-client#52]
Background:
I got my Signet and installed 0.9.6 on Ubuntu and set up the Signet with a passphrase and backups but didn't actually add any entries.
Later I wrote some entries to my Signet using 0.9.9 on a Mac and was able to access them on a couple machines using 0.9.9.

Issue:
I went back to the machine with 0.9.6 and attempted to make a backup as prompted after unlocking, but it failed with an error. After closing out of the backup I was alarmed to discover that none of my entries (all of 2 categories and 2 entries at this point) weren't displaying. I closed and reopened the app but the list was still empty, but I recalled I had gotten a prompt to check for updates and had opened the browser but hadn't yet downloaded the newer version.
I then downloaded and launched the new version to discover to my relief that the entries were now visible.

Another possible issue:
I also noticed when I went to create a backup that it succeeded, BUT the failed backup had actually created a file previously even though it was potentially invalid. I would have expected it to remove the file if the backup wasn't viable, or possibly a post write test that ensures the backup is readable/usable before leaving the file in place.

Desired behavior:
I'm not sure what the planned forward/backwards compatibility is for the entries database, but it would be good to have some kind of a SemVer/API version that would allow older versions to read at least the information it knows about rather than presenting an empty list to the user.

**Comment by nthdimtech** [comment]
The database was designed to be forwards compatible but not backwards compatible. Each entry has a data type and data type version and the client will convert the data to the latest revision as needed. 

Account grouping is a brand new feature so if you assigned each account to a group it would save it in the newest revision of the account data type which would make it inaccessible to earlier client revisions. I think I need to create a special warning when an entry cannot be read because the type revision is too new. Luckily grouping is probably the only feature for quite awhile that will require new data type versions.

I'll create another issue for cleaning up incomplete backups. I thought I had code to delete the file in there but I must have missed something.

**Comment by nthdimtech** [comment]
I've fixed the incomplete backups problem and also added a warning dialog whenever there are entries that cannot be read due to the type version being too new.

I will probably add more backwards compatibility in 1.0+ releases but I have yet to work out mechanisms that would be both reasonably flexible and space efficient.
